### PR TITLE
Thibaut-Refacto-PR

### PR DESCRIPTION
**Pull request by BOPS38IMT**

### CODE REVIEW

**Comment by BOPS38IMT** [comment]
Conflicts resolution summary @maestro-bene 
- application.yml (authentication) : add spring.mail after rabbitmq security conf
- .env : delete my old changes to keep integration ones
- docker-compose.yml : keep integration version with one port ext exposition adding for authentication service (need for mail)
- api.json : keep changes from refacto

File: src/backend/authenticationService/src/main/java/com/example/authenticationservice/model/Role.java
Comment (maestro-bene) [inline src/backend/authenticationService/src/main/java/com/example/authenticationservice/model/Role.java:1]: isEnable => isEnabled is cleaner

File: src/backend/authenticationService/src/main/resources/application.yml
Comment (maestro-bene) [inline src/backend/authenticationService/src/main/resources/application.yml:1]: Port to be in a environment variable; password in docker secret

**Review by maestro-bene — APPROVED** [review]
system files have been changed, there has to be a hot fix after merging to adapt Swarm deployment.
There also has to be a security hotfix after merging for new email capabilities.

**Pull request merged**